src/contextForSubscriptions.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import { useRef, createContext, useContext } from 'react';
+
+type Instance = Symbol;
+
+/**
+ * Returns a unique reference to the calling component
+ * via a Symbol
+ */
+export function useSelf(): Instance {
+  const ref = useRef(Symbol('self'));
+  return ref.current;
+}
```

Comment:
@kcking fixed the self-rerender thing by creating a one-time reference in each hook and then ignoring requests from that hook 👍